File: exercism/python/triangle/triangle.py
def check(sides):
    a = sides[0]
    b = sides[1]
    c = sides[2]
    if a == 0 or b == 0 or c == 0:
        return False
    if a + b <= c or b + c <= a or a + c <= b:
        return False
    return True

def equilateral(sides):
    a = sides[0]
    b = sides[1]
    c = sides[2]
    if not check(sides):
        return False
        
    if a == b == c:
        return True
    else:
        return False


def isosceles(sides):
    a = sides[0]
    b = sides[1]
    c = sides[2]
    if not check(sides):
        return False

    if a == b or a == c or b == c:
        return True
    else:
        return False


def scalene(sides):
    a = sides[0]
    b = sides[1]
    c = sides[2]
    if not check(sides):
        return False
    if a != b and a != c and b != c:
        return True
    else:
        return False

File: exercism/typescript/triangle/triangle.ts
export class Triangle {
  a: number  
  b: number 
  c: number  
  
  constructor(...sides: number[]) {
    this.a = sides[0];
    this.b = sides[1];
    this.c = sides[2];
  }

  get check(): boolean {
    if (this.a === 0 || this.b === 0 || this.c === 0) {
      return false;
    }
    if (this.a + this.b <= this.c || this.b + this.c <= this.a || this.a + this.c <= this.b) {
      return false;
    }
    return true;
  };

  get isEquilateral(): boolean {
    if (!this.check) {
      return false;
    }
    return this.a === this.b && this.b === this.c;
  }

  get isIsosceles(): boolean {
    if (!this.check) {
      return false;
    }
    return this.a === this.b || this.a === this.c || this.b === this.c;
  }

  get isScalene(): boolean {
    if (!this.check) {
      return false;
    }
    return this.a !== this.b && this.a !== this.c && this.b !== this.c;
  }
}


